core/util/src/main/java/org/eclipse/rdf4j/common/io/Sink.java

```diff
@@ -0,0 +1,27 @@
+/******************************************************************************* 
+ * Copyright (c) 2020 Eclipse RDF4J contributors. 
+ * All rights reserved. This program and the accompanying materials 
+ * are made available under the terms of the Eclipse Distribution License v1.0 
+ * which accompanies this distribution, and is available at 
+ * http://www.eclipse.org/org/documents/edl-v10.php. 
+ *******************************************************************************/
+package org.eclipse.rdf4j.common.io;
+
+import org.eclipse.rdf4j.common.lang.FileFormat;
+
+/**
+ * 
+ * A Sink writes data in a particular {@link FileFormat}.
+ * 
+ * @author Jeen Broekstra
+ * @since 3.5.0
+ */
+public interface Sink {
```

Comment:
Fwiw I'm open to other/better naming suggestions. "Sink" is _very_ generic.

Reply:
Thanks, all decent suggestions, but the core problem is that it should cover both RDF data and SPARQL Tuple Query results, so I'm a little uncomfortable prefixing it with "Rdf". I'll go the path of least resistance and stick with Sink for now I think.